optuna/study.py

```diff
@@ -1104,12 +1105,21 @@ def objective(trial):
 
     if len(directions) < 1:
         raise ValueError("The number of objectives must be greater than 0.")
-    elif any(d != "minimize" and d != "maximize" for d in directions):
+    elif any(
+        d not in ["minimize", "maximize", StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE]
+        for d in directions
+    ):
         raise ValueError("Please set either 'minimize' or 'maximize' to direction.")
 
-    direction_objects = [
-        StudyDirection.MINIMIZE if d == "minimize" else StudyDirection.MAXIMIZE for d in directions
-    ]
+    direction_objects = []
+    for d in directions:
+        if d == "minimize":
+            direction_objects.append(StudyDirection.MINIMIZE)
+        elif d == "maximize":
+            direction_objects.append(StudyDirection.MAXIMIZE)
+        else:
+            assert isinstance(d, StudyDirection)
```

Comment:
I've added this assertion to pass the mypy check. Please let me know if there's much smarter way.

Reply:
Not sure this addresses your comment but we could also write something like this.
```python
direction_objects = [d if isintance(d, StudyDirection) else StudyDirection[d.upper()] for d in directions]
```

Reply:
Cool! I think this is much simpler and resolves the redundant assertion.